python_modules/dagster/dagster/_core/definitions/metadata/source_code.py

```diff
@@ -218,11 +240,35 @@ def _build_gitlab_url(url: str, branch: str) -> str:
 
 
 @experimental
+@deprecated(breaking_version="1.8", additional_warn_text="Use `link_to_git` instead.")
 def link_to_source_control(
     assets_defs: Sequence[Union["AssetsDefinition", "SourceAsset", "CacheableAssetsDefinition"]],
     source_control_url: str,
     source_control_branch: str,
     repository_root_absolute_path: Union[Path, str],
+) -> Sequence[Union["AssetsDefinition", "SourceAsset", "CacheableAssetsDefinition"]]:
+    caller_file = Path(inspect.stack()[3].filename).absolute()
+    file_anchor_path_in_repository = os.path.relpath(caller_file, repository_root_absolute_path)
+
+    return link_to_git(
+        assets_defs=assets_defs,
+        source_control_url=source_control_url,
+        source_control_branch=source_control_branch,
+        source_control_file_path_mapping=SourceControlFilePathMapping(
+            local_file_anchor=(caller_file),
+            file_anchor_path_in_repository=file_anchor_path_in_repository,
+        ),
+    )
+
+
+@experimental
+def link_to_git(
+    assets_defs: Sequence[Union["AssetsDefinition", "SourceAsset", "CacheableAssetsDefinition"]],
+    source_control_url: str,
+    source_control_branch: str,
+    source_control_file_path_mapping: Union[
```

Comment:
The fact that we accept this union of `SourceControlFilePathMappingFn` and "object that can be converted to a `SourceControlFilePathMappingFn`" makes me think we should maybe have a simpler parameter that's just something like:

```python

def link_to_git(
    ...
    file_path_mapper: FilePathMapper
) ...

...

class FilePathMapper(ABC):
    @abstractmethod
    def map_file_path(path) -> str: ...

class AnchorBasedFilePathMapper(FilePathMapper):
    ...
```

This would make it a little more onerous in the "custom mapping" case, because you'd need to write a class instead of just a lambda. But arguably that case is pretty fancy already and that's OK?

Thoughts?

Reply:
Hmm, I updated the PR to have

```python

FilePathMappingFn = Callable[[Path], str]

class AnchorBasedFilePathMappingFn(NamedTuple):
  ...
   
  def __call__(self, local_path: Path) -> str: ...
```

That way we still have a single object type but allow lambdas. What do you think?

Reply:
I'm mixed on this. Definitely nice not to force users to define a class to customize this, but feels weird to call something a `Fn` that's not a function.  If we go this route, `AnchorBasedFilePathMapping` or `AnchorBasedFilePathMappingCallable` would be clearer.

I'm curious whether @schrockn has strong instincts here.  @schrockn - ideally it should be possible to glean the necessary context to weigh in from the [docstring](https://github.com/dagster-io/dagster/pull/22353/files#diff-4d1fe96a7ed1f13fc9d851ac4cea2339c7b3792f2ce3da4288cadd90e74dd99fR272), but can explain if more detail in helpful.